### ISSUE TITLE
Make base URI mandatory?

### ISSUE THREAD

**Issue by mlagally** [w3c/wot-profile#153]
It would significantly simplify consumer implementations, if the base URI is mandatory and all form values are relative.

**Comment by benfrancis** [comment]
I agree with @danielpeintner that making `base` mandatory doesn't necessarily simplify a consumer implementation, it just means that every URI has to be resolved against the base as an additional step. Also if a Thing only has a single Form, or two Forms with different origins, then the base may be redundant.

**Comment by mryuichi** [comment]
There are an useful use case when using the base URI. For example, when generating a shadow TD on the gateway, the shadow TD can be created by changing the base URI of the original TD according to the gateway address. Conversely, assigning URIs individually makes it difficult to generate shadow URI.
It's too aggressive to make it mandatory for this reason. However, when using gateway to aggregate devices, it is necessary to generate shadows, so it is recommended (not mandatory)  to design the device TD with this in mind. What about the solution of adding the text that the base URI should be used in this case?

**Comment by mlagally** [comment]
My original thought was that we want to prevent TDs with different absolute URIs in individual affordances, since this may cause a security risk.
Let's discuss in Profile 1.1.
